src/Dialog/Dialog.js

```diff
@@ -349,7 +349,7 @@ class DialogInline extends Component {
           show={open}
           className={overlayClassName}
           style={styles.overlay}
-          onTouchTap={this.handleTouchTapOverlay}
+          onMouseDown={this.handleMouseDownOverlay}
```

Comment:
Yeah, this is the problem section. If it was just for the DatePicker it would be fine, but the Dialog is used by a lot of other components. I wasn't too sure how to handle it. 

The motivation for the change was the fact that the `onTouchTap` triggers the `mouseup` event, which is triggered after the popup appears, which caused the dialog to disappear. I'll have a look again tonight over it and see if there's something silly I've done.

 If you think you know what it could be any guidance would be appreciated.